03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,47 +1,49 @@
 const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/asyncWrapper')
+const { createCustomError } = require('../errors/customError')
 var ObjectId = require('mongoose').Types.ObjectId;
 
 
-const allTasks = async (req, res) => {
+const allTasks = asyncWrapper(async (req, res) => {
     const tasks = await Task.find({})
     res.status(200).json({ tasks })
-}
+})
 
-const newTask = async (req, res) => {
+const newTask = asyncWrapper(async (req, res) => {
     const task = await Task.create(req.body)
-    res.status(201).json({ "Message": 'Succeeded', "data": task })
-}
+    res.status(201).json({ task })
+})
 
-const getTask = async (req, res) => {
+const getTask = asyncWrapper(async (req, res, next) => {
     const id = new ObjectId(req.params.id)
     const task = await Task.findById(id).exec();
     if (!task) {
-        return res.status(404).json({ "message": `No task associated with id: ${id}` })
+        return next(createCustomError(`No task associated with id: ${id}`, 404))
     }
     res.status(200).send({ task })
-}
+})
 
-const updateTask = async (req, res) => {
+const updateTask = asyncWrapper(async (req, res) => {
     const id = new ObjectId(req.params.id)
+
     const task = await Task.findOneAndUpdate({ _id: id }, req.body, {
         new: true,
         runValidators: true,
     })
     if (!task) {
-        return res.status(404).json({ "message": `No task associated with id: ${id}` })
+        return next(createCustomError(`No task associated with id: ${id}`, 404))
     }
     res.status(200).send({ task })
-}
+})
 
-const deleteTask = async (req, res) => {
+const deleteTask = asyncWrapper(async (req, res) => {
```

Comment:
Same about the `next` parameter.
```suggestion
const deleteTask = asyncWrapper(async (req, res, next) => {
```